src/rospack.cpp

```diff
@@ -331,16 +331,13 @@ Rosstackage::crawl(std::vector<std::string> search_path,
 {
   if(!force)
   {
-    if(readCache())
+    // read the cache only once (when search_paths_ is still empty)
+    if(search_paths_.empty() && readCache())
```

Comment:
Indeed, if the current search_path has changed, we should read the corresponding cache. I adapted the code, such that there are the following code paths in Rosstackage::crawl() now:
- crawling is enforced -> just (re)crawl
- current search_path matches previous one and rospack already crawled: nothing todo [new]
- current search_path diverges from previous one: read corresponding cache
  - on success: nothing else todo
  - on failure (no cache or too old): re-crawl
- on re-crawling, set the search_paths_ appropriately